src/App.js

```diff
@@ -1,20 +1,68 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
 
 // components
 import PostList from './components/PostList';
+import Header from './components/Header';
 
-// mock data
-import posts from './data/posts.json';
-import users from './data/users.json';
+// contexts
+import { ThemeConfig, ThemeContext } from './contexts/ThemeContext';
+
+// consts
+const API = 'https://jsonplaceholder.typicode.com';
 
 const App = () => {
+    const [theme, setTheme] = useState('light');
+    const [posts, setPosts] = useState([]);
+    const [users, setUsers] = useState([]);
+    const [error, setError] = useState(undefined);
+
+    const removePost = id => {
+        setPosts(prev => prev.filter(post => post.id !== id));
+    };
+
+    useEffect(() => {
+        axios
+            .get(`${API}/posts`)
```

Comment:
mentioned by Nathan, this could be replaced by [axios default](https://github.com/axios/axios#global-axios-defaults)